File: stepper_mbed/stepper.cpp
#include "mbed.h"

/* Motor Pins for IC in sequence - green, black, blue, red */
/* mBed pins for IC in sequence - p24, p23, p22, p21 */

DigitalOut a(p24);
DigitalOut a_(p23);
DigitalOut b(p22);
DigitalOut b_(p21);

char state_array[] = {0x06, 0x02, 0x0A, 0x08, 0x09, 0x01, 0x05, 0x04};

int main() {
            int idx = 0;
            a = state_array[idx] & (1<<0);
            a_ = state_array[idx] & (1<<1);
            b = state_array[idx] & (1<<2);
            b_ = state_array[idx] & (1<<3);    
    
    while(1) {
        for(int cycle = 0; cycle < 50; cycle++)
        {
            for(int idx=0; idx<8; idx++)
            {
                a = state_array[idx] & (1<<0);
                a_ = state_array[idx] & (1<<1);
                b = state_array[idx] & (1<<2);
                b_ = state_array[idx] & (1<<3);                
                wait(0.01);
            }
        }
        wait(1);
    }
}

File: raspberry_pi/stepper.py
import RPi.GPIO as GPIO
import time

GPIO.setmode(GPIO.BOARD)

coil_A_1_pin = 24
coil_A_2_pin = 23
coil_B_1_pin = 22
coil_B_2_pin = 21

GPIO.setup(coil_A_1_pin,GPIO.OUT)
GPIO.setup(coil_A_2_pin,GPIO.OUT)
GPIO.setup(coil_B_1_pin,GPIO.OUT)
GPIO.setup(coil_B_2_pin,GPIO.OUT)

state_array = [0x06,0x02,0x0A,0x08,0x09,0x01,0x05,0x04]

def rotate(angle, speed = 0.01):
    temp = int(round(angle / 0.9))
    idx = 0
    for cycle in xrange(temp):
        if state_array[idx] & (1<<0):  
            GPIO.output(coil_A_1_pin,GPIO.HIGH)
        else:
            GPIO.output(coil_A_1_pin,GPIO.LOW)

        if state_array[idx] & (1<<1):
            GPIO.output(coil_A_2_pin,GPIO.HIGH)
        else:
            GPIO.output(coil_A_2_pin,GPIO.LOW)

        if state_array[idx] & (1<<2):  
            GPIO.output(coil_B_1_pin,GPIO.HIGH)
        else:
            GPIO.output(coil_B_1_pin,GPIO.LOW)

        if state_array[idx] & (1<<3):
            GPIO.output(coil_B_2_pin,GPIO.HIGH)
        else:
            GPIO.output(coil_B_2_pin,GPIO.LOW)
        if idx == 7:
            idx = 0
        else:
            idx += 1
        time.sleep(speed)

rotate(90)        
